### PR TITLE
Create migrate script to change mocha.opts to [json | js | yml | yaml]

### LINKED ISSUE #3910: Soft deprecate "mocha.opts"
Since v6.0.0 "mocha.opts" is considered "legacy".
* soft deprecate
* provide migration script to configuration file format

### PR DESCRIPTION
**Pull request by wnghdcjfe**
### Description of the Change 
add `mocha migrate` command in cli. 
To make `mocha migrate -file ./mocha.opts -type [yml |  json | js |  yaml]` possible

### Benefits 
Help mocha.opts to .mocharc.json, js, yml, yaml

### Interface
```js
//test - version
node ./mocha/bin/mocha migrate-opts -file ./mocha.opts -type json
//real - version
mocha migrate -file ./mocha.opts -type yml
mocha migrate -file ./mocha.opts -type yaml
mocha migrate -file ./mocha.opts -type json
mocha migrate -file ./mocha.opts -type js
```
and then file create like this
```js
{
 "_": [],
 "reporter": "dot",
 "R": "dot",
 "growl": true,
 "G": true,
 "extension": [
  "mjs",
  "js"
 ]
}
``` 
For your information, I used the existing code for the most safe results. `require('../lib/cli/options.js').loadMochaOpts` so `"_": [],"G": true, ...` lefted but that is no problem with performance.

### Applicable issues 
close #3910

### CODE REVIEW

File: package-scripts.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ function test(testName, mochaParams) {
     mochaParams += ' --forbid-only';
   }
   return `${
-    process.env.COVERAGE ? coverageCommand : ''
+    process.env.COVERsAGE ? coverageCommand : ''
```
Comment (HyunSangHan) [inline package-scripts.js]: I think it might be a typo.

File: README.md [hunk 2]
```diff
@@ -83,12 +83,12 @@ Does your company use Mocha?  Ask your manager or marketing team if your company
 
 You might want to know that:
 
-- Mocha is the *most-depended-upon* module on npm (source: [libraries.io](https://libraries.io/search?order=desc&platforms=NPM&sort=dependents_count)), and
-- Mocha is an *independent* open-source project, maintained exclusively by volunteers.
+- Mocha is the _most-depended-upon_ module on npm (source: [libraries.io](https://libraries.io/search?order=desc&platforms=NPM&sort=dependents_count)), and
+- Mocha is an _independent_ open-source project, maintained exclusively by volunteers.
```
Comment (HyunSangHan) [inline README.md]: Are these changed by markdown-linter automatically?

File: package-scripts.js (hunk 1 above)
Comment (wnghdcjfe) [reply package-scripts.js]: My mistake. I will fix it. thank you.

File: README.md (hunk 2 above)
Comment (wnghdcjfe) [reply README.md]: Yes. I think markdown-linter automatically.. I will fix it. thank you

File: test/migrate-opts/migrate-opts.spec.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      var _path = path.join(process.cwd(), './test/migrate-opts');
+      migrateOpts.init(filepath, type, _path);
+      var __path = path.join(
+        process.cwd(),
+        './test/migrate-opts',
+        '.mocharc.' + type
+      );
+      var isFile = fs.existsSync(__path);
+      expect(isFile).to.equal(true);
+      fs.unlinkSync(__path);
```
Comment (fearphage) [inline test/migrate-opts/migrate-opts.spec.js]: This line will not be run if the previous assertion fails. Clean up in the `after` to ensure the files are removed.

Comment (wnghdcjfe) [reply test/migrate-opts/migrate-opts.spec.js]: Thanks. I will change that code.

**Comment by azu** [comment]
What is the status of this Pull Request?
Is there anything that I can help with? 
(For me, Migration script is blocker to update Mocha@7)

**Comment by wnghdcjfe** [comment]
@azu  I really appreciate your mind once. Based on some feedback from some people,I fixed my code and I think there is no code problems But, I think the maintainers are busy.  so This PR has no progress.

File: scripts/migrate-opts.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  } else if (type === 'js') {
+    writeFile.js(content, _path);
+  } else {
+    writeFile.json(content, _path);
+  }
+  return content;
+};
+
+exports.init = (filepath, type, _path = process.cwd()) => {
+  const content = loadMochaOpts({opts: filepath});
```
[the remark this answers is not in the document]
Comment (azu) [reply scripts/migrate-opts.js]: My patch.

```js
  const content = loadMochaOpts({opts: filepath});
  const types = require("./run-option-metadata").types;
  // filtering by types that remove alias and yargs property
  const allowPropertyNames = Object.values(types).reduce((names, flagNames) => {
    return names.concat(flagNames);
  }, []);
  const normalizedContent = Object.keys(content)
      .filter(key => allowPropertyNames.includes(key))
      .reduce((obj, key) => {
        obj[key] = content[key];
        return obj;
      }, {});
  writeConfig(type, normalizedContent, _path);
```

**Comment by JoshuaKGoldberg** [comment]
👋 coming back to this @wnghdcjfe, are you still interested in working on this PR? As of https://github.com/mochajs/mocha/issues/5027 there are again maintainers who can review it now. No worries if you no longer have time - but if you do that'd be great!

_(I do see that your last message was asking for review - unless directed otherwise we'll try to review it once we're ramped up enough!)_

**Comment by JoshuaKGoldberg** [comment]
Closing out to keep the queue small, as this is a migration for a feature removed 3-4 years ago. Cheers! 🤎

**Comment by wnghdcjfe** [comment]
I just checked this now, I'll join in.

**Pull request closed without merging**